### ISSUE TITLE
wget gives certificate error for dotnet.microsoft.com

### ISSUE THREAD

**Issue by tmds** [dotnet/core#6830]
On Fedora 34:

```
$ wget https://dot.net/v1/dotnet-install.sh
--2021-10-21 14:14:23--  https://dot.net/v1/dotnet-install.sh
Resolving dot.net (dot.net)... 40.112.72.205, 40.113.200.201, 104.215.148.63, ...
Connecting to dot.net (dot.net)|40.112.72.205|:443... connected.
HTTP request sent, awaiting response... 301 Moved Permanently
Location: https://dotnet.microsoft.com/download/dotnet/scripts/v1/dotnet-install.sh [following]
--2021-10-21 14:14:23--  https://dotnet.microsoft.com/download/dotnet/scripts/v1/dotnet-install.sh
Resolving dotnet.microsoft.com (dotnet.microsoft.com)... 13.107.213.67, 13.107.246.67, 2620:1ec:46::67, ...
Connecting to dotnet.microsoft.com (dotnet.microsoft.com)|13.107.213.67|:443... connected.
ERROR: The certificate of ‘dotnet.microsoft.com’ is not trusted.
```

cc @leecow

**Comment by mairaw** [comment]
We had a similar thread before where we couldn't determine why dot.net wasn't working. @ChrisSfanos helped with this before for the dot.net redirection service.

Do you also have issues if you use the dotnet.microsoft.com URL?

**Comment by mairaw** [comment]
Tagging @terrimorton as well

**Comment by ChrisSfanos** [comment]
So we just did an ECR and rotated the certificates.  Do we have a mismatch somewhere?  Is the older cert still in use on FrontDoor?

**Comment by dagood** [comment]
> On a machine that repos, are you able to see the certificate it's grabbing? If so, can you share the first 5 characters of the thumbprint?

Here's what I get:

```
$ echo -n | openssl s_client -connect dotnet.microsoft.com:443 | openssl x509 -fingerprint
depth=2 C = US, O = DigiCert Inc, OU = www.digicert.com, CN = DigiCert Global Root G2
verify return:1
depth=1 C = US, O = Microsoft Corporation, CN = Microsoft Azure TLS Issuing CA 06
verify return:1
depth=0 C = US, ST = WA, L = Redmond, O = Microsoft Corporation, CN = dotnet.microsoft.com
verify return:1
DONE
SHA1 Fingerprint=6A:45:41:56:9D:97:33:C4:4E:B5:1E:FB:BA:F9:45:98:8D:4D:CA:E5
-----BEGIN CERTIFICATE-----
MIIIUDCCBjigAwIBAgITMwAcSpNXY6QTeMcfxgAAABxKkzANBgkqhkiG9w0BAQwF
ADBZMQswCQYDVQQGEwJVUzEeMBwGA1UEChMVTWljcm9zb2Z0IENvcnBvcmF0aW9u
[...]
```

**Comment by ChrisSfanos** [comment]
Thanks - I checked and that certificate is the latest one (rotated into place on 10/7).  

When I visit dotnet.microsoft.com in the browser, that is also the certificate I get the full path is valid.  So I admit I'm not sure what is going on in this case yet

**Comment by tmds** [comment]
@rzio I've added your comment to the gnu_tls bugzilla ticket: https://bugzilla.redhat.com/show_bug.cgi?id=2003363.

**Comment by ueno** [comment]
> RFC 5019 which standardizes "The Lightweight Online Certificate Status Protocol (OCSP) Profile for High-Volume Environments" states that only SHA1 should be supported.

I think this is not relevant; the RFC only states:
> Clients MUST use SHA1 as the hashing algorithm for the CertID.issuerNameHash and the CertID.issuerKeyHash values.

and nothing else, while we are talking about SHA1 usage in _signatures_. Also note that it is a distro-wide decision to disable SHA1 signatures by default, not a decision by the single TLS library; the detailed rationale can be found at the [change proposal](https://fedoraproject.org/wiki/Changes/StrongCryptoSettings2).

**Comment by tmds** [comment]
:/

![screenshot](https://user-images.githubusercontent.com/1025424/146928568-848839e9-1240-4165-8c17-7e07c735e06e.png)

**Comment by ueno** [comment]
That is https://bugzilla.mozilla.org/show_bug.cgi?id=1745600 (and https://bugzilla.mozilla.org/show_bug.cgi?id=966856#c8).

**Comment by BEEDELLROKEJULIANLOCKHART** [comment]
Is addition of '--no-check-certificate' to the script too insecure to be feasible?

**Comment by BEEDELLROKEJULIANLOCKHART** [comment]
More appears to be occuring than merely this. How are you managing to compile PowerShell?

wget --no-check-certificate https://dotnet.microsoft.com/download/dotnet-core/scripts/v1
--2022-03-26 01:53:03--  https://dotnet.microsoft.com/download/dotnet-core/scripts/v1
Resolving dotnet.microsoft.com (dotnet.microsoft.com)... 13.107.246.64, 13.107.213.64, 2620:1ec:bdf::64, ...
Connecting to dotnet.microsoft.com (dotnet.microsoft.com)|13.107.246.64|:443... connected.
WARNING: The certificate of ‘dotnet.microsoft.com’ is not trusted.
HTTP request sent, awaiting response... 302 Found
Cookie coming from dotnet.microsoft.com attempted to set domain to dotnetwebsite.azurewebsites.net
Cookie coming from dotnet.microsoft.com attempted to set domain to dotnetwebsite.azurewebsites.net
Location: /en-US/download/dotnet-core/scripts/v1 [following]
--2022-03-26 01:53:04--  https://dotnet.microsoft.com/en-US/download/dotnet-core/scripts/v1
Reusing existing connection to dotnet.microsoft.com:443.
HTTP request sent, awaiting response... 301 Moved Permanently
Location: /en-us/download/dotnet-core/scripts/v1 [following]
--2022-03-26 01:53:04--  https://dotnet.microsoft.com/en-us/download/dotnet-core/scripts/v1
Reusing existing connection to dotnet.microsoft.com:443.
HTTP request sent, awaiting response... 404 Not Found
Cookie coming from dotnet.microsoft.com attempted to set domain to dotnetwebsite.azurewebsites.net
Cookie coming from dotnet.microsoft.com attempted to set domain to dotnetwebsite.azurewebsites.net
2022-03-26 01:53:04 ERROR 404: Not Found.

**Comment by BEEDELLROKEJULIANLOCKHART** [comment]
This prevents compilation of PowerShell.

**Comment by tmds** [comment]
> Is addition of '--no-check-certificate' to the script too insecure to be feasible?

Yes, this is a no-go. When you add this, you can't consider anything that happens next to be secure/trusted.

**Comment by rzio** [comment]
Update from the OneOCSP side: Due to a new CAB forum requirement, OneOCSP will switch the algorithm to SHA-256 by 2022-5-31.
